x-pack/solutions/search/plugins/search_playground/public/components/query_mode/query_output.tsx

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+import {
+  EuiButton,
+  EuiEmptyPrompt,
+  EuiFlexGroup,
+  EuiSplitPanel,
+  EuiText,
+  useEuiTheme,
+} from '@elastic/eui';
+import { CodeEditor } from '@kbn/code-editor';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { FullHeight, QueryViewTitlePanel } from './styles';
+import { QueryTestResponse } from '../../types';
+
+export interface ElasticsearchQueryOutputProps {
+  queryResponse?: QueryTestResponse;
+  queryError?: unknown;
+  isError: boolean;
+  isLoading: boolean;
+  executeQuery: () => void;
+}
+
+export const ElasticsearchQueryOutput = ({
+  queryResponse,
+  executeQuery,
+  isError,
+  isLoading,
+  queryError,
+}: ElasticsearchQueryOutputProps) => {
+  const { euiTheme } = useEuiTheme();
+  const respJSON = useMemo(() => {
+    if (isError) {
```

Comment:
non-blocking: ideally this would be a utility function

Reply:
I wrote one for the backend and I will re-use it here. We may end up wanting to handle those slightly differently in the future but we'll see.